Vala-Language-Server fails to install (regex issue fixed in master)

### I've searched open issues for similar requests

Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

Vala-Language-Server fails to install, this stems from a [bug](https://github.com/vala-lang/vala-language-server/issues/263) that was fixed in this [PR](https://github.com/vala-lang/vala-language-server/pull/268), however there hasn't been a tagged release since this was fixed, so the one Mason uses still has the bug. A fix could be to just use the master branch and build that from source! :) 

Thanks,
Amaan

### Expected behavior

It installs

### Affected packages

vala-language-server

### Mason output

```Text
Failed
    ◍ vala-language-server
      ▼ Displaying full log
        Downloading file "https://github.com/vala-lang/vala-language-server/releases/download/0.48.5/vala-language-server-0.48.5.tar.xz"...
        vala-language-server-0.48.5/
        vala-language-server-0.48.5/.editorconfig
        vala-language-server-0.48.5/.github/
        vala-language-server-0.48.5/.github/FUNDING.yml
        vala-language-server-0.48.5/.github/ISSUE_TEMPLATE/
        vala-language-server-0.48.5/.github/ISSUE_TEMPLATE/bug_report.md
        vala-language-server-0.48.5/.github/workflows/
        vala-language-server-0.48.5/.github/workflows/vala-daily.yml
        vala-language-server-0.48.5/.github/workflows/vala-lts.yml
        vala-language-server-0.48.5/.gitignore
        vala-language-server-0.48.5/.vala-lint.conf
        vala-language-server-0.48.5/COPYING
        vala-language-server-0.48.5/README.md
        vala-language-server-0.48.5/config.vala.in
        vala-language-server-0.48.5/data/
        vala-language-server-0.48.5/data/man/
        vala-language-server-0.48.5/data/man/vala-language-server.1.scd
        vala-language-server-0.48.5/data/meson.build
        vala-language-server-0.48.5/images/
        vala-language-server-0.48.5/images/vls-gb.png
        vala-language-server-0.48.5/images/vls-vim.png
        vala-language-server-0.48.5/images/vls-vscode.png
        vala-language-server-0.48.5/lint.sh
        vala-language-server-0.48.5/meson.build
        vala-language-server-0.48.5/meson_options.txt
        vala-language-server-0.48.5/plugins/
        vala-language-server-0.48.5/plugins/gnome-builder/
        vala-language-server-0.48.5/plugins/gnome-builder/get_builder_abi.sh
        vala-language-server-0.48.5/plugins/gnome-builder/meson.build
        vala-language-server-0.48.5/plugins/gnome-builder/vala_langserv.plugin.in
        vala-language-server-0.48.5/plugins/gnome-builder/vala_langserv.py
        vala-language-server-0.48.5/src/
        vala-language-server-0.48.5/src/analysis/
        vala-language-server-0.48.5/src/analysis/codeanalyzer.vala
        vala-language-server-0.48.5/src/analysis/codelensanalyzer.vala
        vala-language-server-0.48.5/src/analysis/codestyleanalyzer.vala
        vala-language-server-0.48.5/src/analysis/inlayhintnodes.vala
        vala-language-server-0.48.5/src/analysis/symbolenumerator.vala
        vala-language-server-0.48.5/src/codeaction/
        vala-language-server-0.48.5/src/codeaction/baseconverteraction.vala
        vala-language-server-0.48.5/src/codeaction/implementmissingprereqsaction.vala
        vala-language-server-0.48.5/src/codehelp/
        vala-language-server-0.48.5/src/codehelp/callhierarchy.vala
        vala-language-server-0.48.5/src/codehelp/codeaction.vala
        vala-language-server-0.48.5/src/codehelp/codehelp.vala
        vala-language-server-0.48.5/src/codehelp/codelensengine.vala
        vala-language-server-0.48.5/src/codehelp/completionengine.vala
        vala-language-server-0.48.5/src/codehelp/find_scope.vala
        vala-language-server-0.48.5/src/codehelp/formatter.vala
        vala-language-server-0.48.5/src/codehelp/nodesearch.vala
        vala-language-server-0.48.5/src/codehelp/signaturehelpengine.vala
        vala-language-server-0.48.5/src/codehelp/symbolextractor.vala
        vala-language-server-0.48.5/src/codehelp/symbolreferences.vala
        vala-language-server-0.48.5/src/codehelp/symbolvisitor.vala
        vala-language-server-0.48.5/src/codehelp/typehierarchy.vala
        vala-language-server-0.48.5/src/documentation/
        vala-language-server-0.48.5/src/documentation/cnamemapper.vala
        vala-language-server-0.48.5/src/documentation/doccomment.vala
        vala-language-server-0.48.5/src/documentation/girdocumentation.vala
        vala-language-server-0.48.5/src/meson.build
        vala-language-server-0.48.5/src/projects/
        vala-language-server-0.48.5/src/projects/buildtarget.vala
        vala-language-server-0.48.5/src/projects/buildtask.vala
        vala-language-server-0.48.5/src/projects/ccproject.vala
        vala-language-server-0.48.5/src/projects/compilation.vala
        vala-language-server-0.48.5/src/projects/defaultproject.vala
        vala-language-server-0.48.5/src/projects/filecache.vala
        vala-language-server-0.48.5/src/projects/mesonproject.vala
        vala-language-server-0.48.5/src/projects/project.vala
        vala-language-server-0.48.5/src/projects/textdocument.vala
        vala-language-server-0.48.5/src/projects/types.vala
        vala-language-server-0.48.5/src/protocol.vala
        vala-language-server-0.48.5/src/reporter.vala
        vala-language-server-0.48.5/src/request.vala
        vala-language-server-0.48.5/src/server.vala
        vala-language-server-0.48.5/src/util.vala
        vala-language-server-0.48.5/src/windows.vapi
        vala-language-server-0.48.5/subprojects/
        vala-language-server-0.48.5/subprojects/gobject-list-prince781.wrap
        vala-language-server-0.48.5/subprojects/json-glib.wrap
        vala-language-server-0.48.5/subprojects/jsonrpc-glib.wrap
        vala-language-server-0.48.5/subprojects/libgee.wrap
        vala-language-server-0.48.5/test/
        vala-language-server-0.48.5/test/meson.build
        vala-language-server-0.48.5/test/testclient.vala
        vala-language-server-0.48.5/vapi/
        vala-language-server-0.48.5/vapi/jsonrpc-glib-1.0.vapi
        vala-language-server-0.48.5/version.vala.in
        The Meson build system
        Version: 1.0.0
        Source dir: /home/amaanq/.local/share/nvim/mason/.packages/vala-language-server/vala-language-server-0.48.5
        Build dir: /home/amaanq/.local/share/nvim/mason/.packages/vala-language-server/vala-language-server-0.48.5/build
        Build type: native build
        Project name: vala-language-server
        Project version: 0.48.5
        C compiler for the host machine: sccache cc (gcc 12.2.0 "cc (GCC) 12.2.0")
        C linker for the host machine: cc ld.bfd 2.39.0
        Vala compiler for the host machine: valac (valac 0.56.3)
        Host machine cpu family: x86_64
        Host machine cpu: x86_64
        WARNING: You should add the boolean check kwarg to the run_command call.
                 It currently defaults to false,
                 but it will default to true in future releases of meson.
                 See also: https://github.com/mesonbuild/meson/issues/9300
        Found pkg-config: /usr/bin/pkg-config (1.8.0)
        Run-time dependency libvala-0.56 found: YES 0.56.3
        Run-time dependency gobject-2.0 found: YES 2.74.4
        Run-time dependency jsonrpc-glib-1.0 found: YES 3.42.0
        Run-time dependency glib-2.0 found: YES 2.74.4
        Run-time dependency gio-2.0 found: YES 2.74.4
        Run-time dependency gee-0.8 found: YES 0.20.6
        Run-time dependency json-glib-1.0 found: YES 1.6.6
        Library posix found: YES
        Configuring config.vala using configuration
        Run-time dependency gio-unix-2.0 found: YES 2.74.4
        Found pkg-config: /usr/bin/pkg-config (1.8.0)
        Build-time dependency scdoc found: YES 1.11.2
        Program /usr/bin/scdoc found: YES (/usr/bin/scdoc)
        Program sh found: YES (/usr/bin/sh)
        Configuring vala-language-server.1 with command
        Program sh found: YES (/usr/bin/sh)
        Program sed found: YES (/usr/bin/sed)
        Program gnome-builder found: NO
        Build targets in project: 3
        
        vala-language-server 0.48.5
        
          User defined options
            prefix: /home/amaanq/.local/share/nvim/mason/.packages/vala-language-server
        
        Found ninja-1.11.1 at /usr/bin/ninja
        WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.
        ninja: Entering directory `build'
        [1/48] Generating version.vala with a custom command
        fatal: not a git repository (or any of the parent directories): .git
        [2/48] Compiling Vala source ../test/testclient.vala ../src/util.vala
        [3/48] Compiling C object test/vls-testclient.p/meson-generated_testclient.c.o
        [4/48] Compiling C object test/vls-testclient.p/meson-generated_util.c.o
        [5/48] Linking target test/vls-testclient
        [6/48] Compiling Vala source ../src/analysis/codeanalyzer.vala ../src/analysis/codelensanalyzer.vala ../src/analysis/codestyleanalyzer.vala ../src/analysis/inlayhintnodes.vala ../src/analysis/symbolenumerator.vala ../src/codeaction/baseconverteraction.vala ../src/codeaction/implementmissingprereqsaction.vala ../src/codehelp/callhierarchy.vala ../src/codehelp/codeaction.vala ../src/codehelp/codehelp.vala ../src/codehelp/codelensengine.vala ../src/codehelp/completionengine.vala ../src/codehelp/find_scope.vala ../src/codehelp/formatter.vala ../src/codehelp/nodesearch.vala ../src/codehelp/signaturehelpengine.vala ../src/codehelp/symbolextractor.vala ../src/codehelp/symbolreferences.vala ../src/codehelp/symbolvisitor.vala ../src/codehelp/typehierarchy.vala ../src/documentation/cnamemapper.vala ../src/documentation/doccomment.vala ../src/documentation/girdocumentation.vala ../src/projects/buildtarget.vala ../src/projects/buildtask.vala ../src/projects/ccproject.vala ../src/projects/compilation.vala ../src/projects/defaultproject.vala ../src/projects/filecache.vala ../src/projects/mesonproject.vala ../src/projects/project.vala ../src/projects/textdocument.vala ../src/projects/types.vala ../src/protocol.vala ../src/reporter.vala ../src/request.vala ../src/server.vala config.vala ../src/util.vala version.vala
        FAILED: src/vala-language-server.p/analysis/codeanalyzer.c src/vala-language-server.p/analysis/codelensanalyzer.c src/vala-language-server.p/analysis/codestyleanalyzer.c src/vala-language-server.p/analysis/inlayhintnodes.c src/vala-language-server.p/analysis/symbolenumerator.c src/vala-language-server.p/codeaction/baseconverteraction.c src/vala-language-server.p/codeaction/implementmissingprereqsaction.c src/vala-language-server.p/codehelp/callhierarchy.c src/vala-language-server.p/codehelp/codeaction.c src/vala-language-server.p/codehelp/codehelp.c src/vala-language-server.p/codehelp/codelensengine.c src/vala-language-server.p/codehelp/completionengine.c src/vala-language-server.p/codehelp/find_scope.c src/vala-language-server.p/codehelp/formatter.c src/vala-language-server.p/codehelp/nodesearch.c src/vala-language-server.p/codehelp/signaturehelpengine.c src/vala-language-server.p/codehelp/symbolextractor.c src/vala-language-server.p/codehelp/symbolreferences.c src/vala-language-server.p/codehelp/symbolvisitor.c src/vala-language-server.p/codehelp/typehierarchy.c src/vala-language-server.p/documentation/cnamemapper.c src/vala-language-server.p/documentation/doccomment.c src/vala-language-server.p/documentation/girdocumentation.c src/vala-language-server.p/projects/buildtarget.c src/vala-language-server.p/projects/buildtask.c src/vala-language-server.p/projects/ccproject.c src/vala-language-server.p/projects/compilation.c src/vala-language-server.p/projects/defaultproject.c src/vala-language-server.p/projects/filecache.c src/vala-language-server.p/projects/mesonproject.c src/vala-language-server.p/projects/project.c src/vala-language-server.p/projects/textdocument.c src/vala-language-server.p/projects/types.c src/vala-language-server.p/protocol.c src/vala-language-server.p/reporter.c src/vala-language-server.p/request.c src/vala-language-server.p/server.c src/vala-language-server.p/config.c src/vala-language-server.p/util.c src/vala-language-server.p/version.c 
        valac -C --debug --debug --define=WITH_JSONRPC_GLIB_3_30 --target-glib=auto --enable-gobject-tracing --fatal-warnings --pkg gio-unix-2.0 --pkg posix --pkg libvala-0.56 --pkg jsonrpc-glib-1.0 --pkg json-glib-1.0 --pkg gee-0.8 --pkg gio-2.0 --pkg gobject-2.0 --pkg glib-2.0 --color=always --directory src/vala-language-server.p --basedir ../src ../src/analysis/codeanalyzer.vala ../src/analysis/codelensanalyzer.vala ../src/analysis/codestyleanalyzer.vala ../src/analysis/inlayhintnodes.vala ../src/analysis/symbolenumerator.vala ../src/codeaction/baseconverteraction.vala ../src/codeaction/implementmissingprereqsaction.vala ../src/codehelp/callhierarchy.vala ../src/codehelp/codeaction.vala ../src/codehelp/codehelp.vala ../src/codehelp/codelensengine.vala ../src/codehelp/completionengine.vala ../src/codehelp/find_scope.vala ../src/codehelp/formatter.vala ../src/codehelp/nodesearch.vala ../src/codehelp/signaturehelpengine.vala ../src/codehelp/symbolextractor.vala ../src/codehelp/symbolreferences.vala ../src/codehelp/symbolvisitor.vala ../src/codehelp/typehierarchy.vala ../src/documentation/cnamemapper.vala ../src/documentation/doccomment.vala ../src/documentation/girdocumentation.vala ../src/projects/buildtarget.vala ../src/projects/buildtask.vala ../src/projects/ccproject.vala ../src/projects/compilation.vala ../src/projects/defaultproject.vala ../src/projects/filecache.vala ../src/projects/mesonproject.vala ../src/projects/project.vala ../src/projects/textdocument.vala ../src/projects/types.vala ../src/protocol.vala ../src/reporter.vala ../src/request.vala ../src/server.vala config.vala ../src/util.vala version.vala
        ../src/documentation/girdocumentation.vala:359.29-359.57: error: Invalid regular expression `//!\[(.*?)\]\(([~:\/\\\w-.]+)\)'.
          359 |             comment_data = /!\[(.*?)\]\(([~:\/\\\w-.]+)\)/
              |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
        ../src/documentation/girdocumentation.vala:389.25-389.47: error: Invalid regular expression `//\[(.*?)\]\[([\w-\s]+)\]'.
          389 |         comment_data = /\[(.*?)\]\[([\w-\s]+)\]/
              |                         ^~~~~~~~~~~~~~~~~~~~~~~ 
        ../src/documentation/doccomment.vala:132.17-132.59: error: Invalid regular expression `//(\[\[|{{)([~:\/\\\w-.]+)(\|(.*?))?(\]\]|}})'.
          132 |         body = /(\[\[|{{)([~:\/\\\w-.]+)(\|(.*?))?(\]\]|}})/
              |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
        Compilation failed: 3 error(s), 0 warning(s)
        ninja: build stopped: subcommand failed.
        spawn: ninja failed with exit code 1 and signal 0.
```


### Installation log

```Text
Same as Mason output?
```


### Neovim version (>= 0.7)

NVIM v0.9.0-dev-588+g99cf11128
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

### Operating system/version

Linux archmain 6.2.0-rc2-1-git-00010-g69b41ac87e4a #1 SMP PREEMPT_DYNAMIC Tue, 03 Jan 2023 12:51:18 +0000 x86_64 GNU/Linux

### Healthcheck

```Text
==============================================================================
mason: require("mason.health").check()

mason.nvim report ~
- OK neovim version >= 0.7.0
- OK **Go**: `go version go1.19.4 linux/amd64`
- OK **cargo**: `cargo 1.68.0-nightly (2381cbdb4 2022-12-23)`
- OK **luarocks**: `/usr/bin/luarocks 3.9.2`
- OK **Ruby**: `ruby 3.0.4p208 (2022-04-12 revision 3fa771dded) [x86_64-linux]`
- OK **RubyGem**: `3.3.25`
- OK **Composer**: `Composer version 2.5.1 2022-12-22 15:33:54`
- OK **PHP**: `PHP 8.1.13 (cli) (built: Dec 12 2022 23:45:33) (NTS)`
- OK **npm**: `8.19.2`
- OK **node**: `v16.19.0`
- OK **python3**: `Python 3.10.9`
- OK **pip3**: `pip 22.3.1 from /home/amaanq/.local/lib/python3.10/site-packages/pip (python 3.10)`
- OK **javac**: `javac 17.0.5`
- OK **java**: `openjdk version "17.0.5" 2022-10-18`
- OK **julia**: `julia version 1.8.4`
- OK **wget**: `GNU Wget 1.21.3 built on linux-gnu.`
- OK **curl**: `curl 7.87.0 (x86_64-pc-linux-gnu) libcurl/7.87.0 OpenSSL/3.0.7 zlib/1.2.13 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.4 libpsl/0.21.1 (+libidn2/2.3.4) libssh2/1.10.0 nghttp2/1.51.0`
- OK **gzip**: `gzip 1.12`
- OK **tar**: `tar (GNU tar) 1.34`
- WARNING **pwsh**: not available
- OK **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
- OK **sh**: `Ok`
- OK GitHub API rate limit. Used: 6. Remaining: 4994. Limit: 5000. Reset: Tue Jan  3 23:48:12 2023.
```


### Screenshots

_No response_

--- a/lua/mason-registry/vala-language-server/init.lua
+++ b/lua/mason-registry/vala-language-server/init.lua
@@ -1,8 +1,10 @@
 local Pkg = require "mason-core.package"
 local std = require "mason-core.managers.std"
 local github = require "mason-core.managers.github"
+local git = require "mason-core.managers.git"
 local platform = require "mason-core.platform"
 local path = require "mason-core.path"
+local Optional = require "mason-core.optional"
 
 return Pkg.new {
     name = "vala-language-server",
@@ -17,21 +19,14 @@ return Pkg.new {
         std.ensure_executable("ninja", { help_url = "https://ninja-build.org/" })
         std.ensure_executable("valac", { help_url = "https://wiki.gnome.org/Projects/Vala" })
 
-        local release_source = github.untarxz_release_file {
-            repo = "vala-lang/vala-language-server",
-            asset_file = function(version)
-                return ("vala-language-server-%s.tar.xz"):format(version)
-            end,
-        }
-        release_source.with_receipt()
+        local repo = "vala-lang/vala-language-server"
+        local source = github.tag { repo = repo }
+        source.with_receipt()
+        git.clone { ("https://github.com/%s.git"):format(repo), version = Optional.of(source.tag) }
 
-        local vala_dirname = ("vala-language-server-%s"):format(release_source.release)
         local install_dir = ctx.cwd:get()
-        ctx:chdir(vala_dirname, function()
-            ctx.spawn.meson { ("-Dprefix=%s"):format(install_dir), "build" }
-            ctx.spawn.ninja { "-C", "build", "install" }
-        end)
-        ctx.fs:rmrf(vala_dirname)
+        ctx.spawn.meson { ("-Dprefix=%s"):format(install_dir), "build" }
+        ctx.spawn.ninja { "-C", "build", "install" }
         ctx:link_bin(
             "vala-language-server",
             path.concat { "bin", platform.is.win and "vala-language-server.exe" or "vala-language-server" }

